File: Linked List/21_merge_two_sorted_lists.js
var mergeTwoLists = function(list1, list2) {
    let new_list = new ListNode();
    let prev_head = new_list;

    while(list1 && list2){
        if(list1.val < list2.val){
            new_list.next = list1;
            list1 = list1.next;
        } else {    
            new_list.next = list2;
            list2 = list2.next;
        }

        new_list = new_list.next;
    }

    new_list.next = list1 || list2;

    return prev_head.next;
};

File: Linked List/21_merge_two_sorted_lists.cs
public class Solution
{
    public ListNode MergeTwoLists(ListNode list1, ListNode list2)
    {
        ListNode list3 = new ListNode();
        ListNode aux = list3;

        while (list1 != null && list2 != null)
        {
            if (list1.val < list2.val)
            {
                list3.next = list1;
                list1 = list1.next;
            }
            else
            {
                list3.next = list2;
                list2 = list2.next;

            }

            list3 = list3.next;
        }

        list3.next = list1 == null ? list2 : list1;

        return aux.next;
    }
}